owledge/spec/models/note_spec.rb

```diff
@@ -0,0 +1,247 @@
+require 'rails_helper'
+
+RSpec.describe Note, type: :model do
+  describe 'check if Only parents function is working as expected' do
+    context 'No records so count is 0' do
+      before :each do
+        Note.delete_all
+      end
+      it 'Checks the list of parents' do
+        @notes = Note.allParents
+        expect(@notes.count).to eq(0)
+      end
+    end
+    context 'Only parents so count of function is equal to count of all' do
+      before :each do
+        Note.delete_all
+        Note.create([{
+                              title: 'Graph Theory',
+                              description: %w{
+                                In mathematics, graph theory is the study of graphs, which are mathematical structures
+                                used to model pairwise relations between objects.
+                              }.join(" ")
+                            }])
+        Note.create([{
+                              title: 'Sorting Algorithms',
+                              description: %w{
+                                In computer science, a sorting algorithm is an algorithm that puts elements of a list in a certain order.
+                              }.join(" ")
+                            }])
+        Note.create([{
+                              title: 'Sorting Algorithms 3',
+                              description: %w{
+                                In computer science, a sorting algorithm is an algorithm that puts elements of a list in a certain order.
+                              }.join(" ")
+                            }])
```

Comment:
You don't need to individually create every Note with an array of objects. Either remove the array or combine them.

Example:

```
Note.create([{
// note 1
}, {
// note 2
}])
```

Reply:
Also - shouldn't this be part of "seeds". Can't we just import seeds here?

Reply:
here: https://stackoverflow.com/a/19930700

Reply:
Cant always be a part of the seed but moved most conditions to seed